test/cxx/Core/ApplicationPool/PoolTest.cpp

```diff
@@ -632,6 +631,66 @@ namespace tut {
 		ensure_equals(pool->getGroupCount(), 0u);
 	}
 
+	TEST_METHOD(15) {
+	    // Test that the process generation increments when the group restarts
+		Options options = createOptions();
+
+		// Spawn a process and opens a session with it.
+		pool->setMax(1);
+		pool->asyncGet(options, callback);
+		EVENTUALLY(5,
+			result = number == 1;
+		);
+
+		// Close the session so that the process is now idle.
+		ProcessPtr process = currentSession->getProcess()->shared_from_this();
+		currentSession.reset();
+		unsigned int gen1 = process->generation;
+
+		ensure(pool->restartGroupByName(options.appRoot));
+		EVENTUALLY(5,
```

Comment:
I don't think this is the right "eventually" condition to check for. You should check that the PID of the sole process in the pool has changed.